integration_tests/run_pyspark_from_build.sh

```diff
@@ -443,6 +444,109 @@ else
         grep 'WARN RapidsPluginUtils: RAPIDS Accelerator is in explain only mode' <<< "$output"
         grep -F 'res0: Array[org.apache.spark.sql.Row] = Array([4950])' <<< "$output"
         echo "SUCCESS explainOnly mode on CPU smoke test"
+    elif [[ "${SPARK_CONNECT_SMOKE_TEST}" != "0" ]]; then
+        echo "Running Spark Connect smoke test..."
+        # Gate on Spark version (3.5.6+ has Connect support with external jars. Example:plugin jar)
+        # https://github.com/apache/spark/pull/50475
+        if [[ "$VERSION_STRING" < "3.5.6" ]]; then
+            echo "SKIPPING Spark Connect smoke test - requires Spark 3.5.6+ but found $VERSION_STRING"
+            exit 0
+        fi
+
+        # Build Connect packages and server-side jars (RAPIDS plugin)
+        CONNECT_PACKAGES="org.apache.spark:spark-connect_${SCALA_VERSION}:${VERSION_STRING}"
+        SERVER_JARS=""
+        if [[ -n "$PYSP_TEST_spark_jars" ]]; then
+            SERVER_JARS="$PYSP_TEST_spark_jars"
+        elif [[ -n "$ALL_JARS" ]]; then
+            SERVER_JARS="${ALL_JARS//:/,}"
+        fi
+
+        # Helper: check if port is listening
+        check_port() {
+            local port=$1
+            if command -v ss >/dev/null 2>&1; then
+                ss -ltn 2>/dev/null | grep -q ":${port} "
+            else
+                netstat -ltn 2>/dev/null | grep -q ":${port} "
+            fi
+        }
+
+        # Pick a free localhost port
+        pick_free_port() {
+            local port
+            for i in $(seq 1 50); do
+                port=$(( (RANDOM % 10000) + 20000 ))
+                if ! check_port "$port"; then
+                    echo "$port"
+                    return 0
+                fi
+            done
+            echo 15002
+        }
+
+        # Prefer default Connect port; if busy, fall back to a free ephemeral port
+        CONNECT_HOST="${CONNECT_HOST:-127.0.0.1}"
+        CONNECT_PORT=15002
+        if check_port "$CONNECT_PORT"; then
+            CONNECT_PORT=$(pick_free_port)
+        fi
+        CONNECT_SERVER_URL="sc://${CONNECT_HOST}:${CONNECT_PORT}"
+
+        cleanup_connect_server() {
+            if [[ -f "${SPARK_HOME}/sbin/stop-connect-server.sh" ]]; then
```

Comment:
non-blocking, and not sure if  it would take less code to run with SPARK_NO_DAEMONIZE  with & to control the child processes ?

Reply:
Thanks for the suggestion @gerashegalov . I tried using SPARK_NO_DAEMONIZE  and it actually added code (PID management, liveness checks, background log capture)  along with the existing readiness probe. The current approach of  start plus stop script keeps things minimal.